src/gui/remote/RemoteHandler.cpp

```diff
@@ -143,3 +134,27 @@ RemoteHandler::RemoteResult RemoteHandler::upload(const QString& filePath, const
         return result;
     });
 }
+
+QString RemoteHandler::getTempFileLocation()
```

Comment:
There's zero error checking done in this function. Both `QDir::mkdir` and `QFile::setPermissions` can return false and the existence of the directory isn't tested prior to creating it regardless how unlikely a collision is.

It's fine for the whole function to just fail if something fails and return an error to the user because it's very unlikely to fail so no retries are necessary.

Reply:
I have added checks to fail in case `QDir::mkdir` and `QFile::setPermissions` return false, although it is unlikely that either of them will return false. `QDir::mkdir` will also return false in case the directory already exists, so no additional existence check is needed.